modules/storages/app/services/storages/one_drive_managed_folder_sync_service.rb

```diff
@@ -223,11 +223,11 @@ def active_project_storages_scope
     end
 
     def client_remote_identities_scope
-      RemoteIdentity.includes(:user).where(oauth_client: @storage.oauth_client)
+      RemoteIdentity.includes(:user).where(integration: @storage, auth_source: @storage.oauth_client)
     end
 
     def admin_remote_identities_scope
-      RemoteIdentity.includes(:user).where(oauth_client: @storage.oauth_client, user: User.admin.active)
+      RemoteIdentity.includes(:user).where(integration: @storage, auth_source: @storage.oauth_client, user: User.admin.active)
```

Comment:
:red_circle: See `nextcloud_managed_folder_sync_service.rb`.

Reply:
Yes, the same.

Reply:
We still need to drop the `auth_source` here. Only `integration` and `user` are enough to identify which users need to be synced.